path_role.go

```diff
@@ -463,19 +463,23 @@ func (b *jwtAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logical.
 	}
 
 	// disregard the trailing slash (if it exists) on all bound audiences if the flag is set
-	if _, ok := data.GetOk("bound_audience_disregard_trailing_slash"); ok {
+	if _, ok := data.GetOk("normalize_bound_audiences"); ok {
 		boundAudiences := []string{}
 		processed := map[string]bool{} // used to prevent duplicate entries
 
 		for _, audience := range role.BoundAudiences {
+			normalizedAudience := audience
+
 			// trim the trailing slash from the audience if it exists
-			audienceWithoutTrailingSlash := strings.TrimRight(audience, "/")
+			if strings.HasSuffix(normalizedAudience, "/") {
+				normalizedAudience = strings.TrimSuffix(normalizedAudience, "/")
+			}
```

Comment:
We probably don't need the `if` check here now, just calling TrimSuffix should suffice?

```suggestion
			normalizedAudience = strings.TrimSuffix(normalizedAudience, "/")
```